### PR TITLE
Feature 6: Filter todo

### PR DESCRIPTION
**Pull request by saa-wtag**
[Task-7: Filter Todo](https://trello.com/c/j1N4AiqF)
This PR introduces the functionality of filter the todos by clicking the all, complete, incomplete buttons.

**Functionalities:**

- Todos can be filtered by completed/ incomplete / all.
- Todos can be filtered among the searched items or vice versa.

Please have a close look on this PR and kindly put your valuable comments.

https://github.com/saa-wtag/ToDoApp_VanillaJS/assets/164306406/d63efa49-5963-425f-9ece-7a25d18bddf3

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -16,22 +19,27 @@ import {
...
+  setActiveButton,
 } from "./utilities.js";
 import { MESSAGES } from "./const.js";
 
 let tasks = [];
 let isVisible = false;
+let currentFilter = "all";
+let filteredOrSearchableTasks = [];
+let tasksDisplayed = 0;
+const tasksPerPage = 9;
```
Comment (rza-wtag) [inline js/script.js]: Move these constants to another `const.js` file and then export it

Comment (saa-wtag) [reply js/script.js]: `tasksPerPage` constant moved to const.js but rest of the things needs to be in this file.

**Review by sal-wtag — APPROVED** [review]

File: js/script.js [hunk 2]
```diff
@@ -43,12 +50,15 @@ const handleTaskView = () => {
   $noTask.style.display = "none";
 
   toggleInputContainer(isVisible, handleAddTask);
-  if (!isVisible) renderTasks(tasks);
+  if (!isVisible)
+    renderTasks(
+      filteredOrSearchableTasks.length ? filteredOrSearchableTasks : tasks
```
Comment (maa-wtag) [inline js/script.js]: please make it a const and reuse it as it is being used in several places

**Review by maa-wtag — APPROVED** [review]

File: js/script.js (hunk 2 above)
Comment (saa-wtag) [reply js/script.js]: Done, bhai.

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**